core/org.wso2.carbon.user.core/src/test/java/org/wso2/carbon/user/core/util/DatabaseUtilTest.java

```diff
@@ -0,0 +1,265 @@
+        String driverName = "org.h2.Driver";
+        PoolProperties poolProperties = new PoolProperties();
+        poolProperties.setUrl(url);
+        poolProperties.setDriverClassName(driverName);
+        DataSource fake = new org.apache.tomcat.jdbc.pool.DataSource(poolProperties);
+
+        Assert.assertNotNull(DatabaseUtil.getDBConnection(fake));
+    }
+
+    @Test
+    public void closeConnectionShouldNotBeNull() throws SQLException {
+
+        Mockito.doAnswer(RETURNS_MOCKS).when(datasource).getConnection();
+        DatabaseUtil.closeConnection(conn);
+        assertNotNull(conn);
+    }
+
+    @Test
+    public void closeConnectionShouldHandlesSQLRecoverableException() throws Exception {
+
+            Mockito.doThrow(new SQLRecoverableException("throw SQLRecoverableException")).when(conn).close();
+            DatabaseUtil.closeConnection(conn);
+            System.out.println("Expected error thrown in sqlerecoverable handler");
+            Assert.assertNotNull(conn);
+    }
+
+    @Test
+    public void newCloseStatementsShouldBBeSameResult() throws SQLException {
+        when(conn.isClosed()).thenReturn(false);
+        DatabaseUtil.close(conn, preparedStatement);
+    }
+
+    @Test
+    public void newClosePreparedStatementShouldHandlesSQLRecoverableException() {
+
+        boolean isThrown = false;
+        try {
+            Mockito.doThrow(new SQLRecoverableException("throw SQLRecoverableException")).when(preparedStatement).close();
+            DatabaseUtil.close(conn, preparedStatement);
+        } catch (SQLRecoverableException e) {
+            isThrown = true;
+            Assert.assertTrue(isThrown);
+        } catch (SQLException e) {
+            // this shouldn't happen
+            Assert.assertTrue(false);
+        }
+        Assert.assertNotNull(preparedStatement);
+    }
+
+    @Test
+    public void testNewCloseResultSetHandlesSQLRecoverableException() {
+
+        try {
+            Mockito.doThrow(new SQLRecoverableException("throw SQLRecoverableException")).when(resultSetMock).close();
+            DatabaseUtil.close(conn, resultSetMock, preparedStatement);
+
+        } catch (SQLException e) {
+            log.error("Unexpected error thrown in sqlerecoverable handler" +  e.getMessage(), e);
+        }
+    }
+    @Test
+    public void testCloseStatements() throws SQLException {
+
+        when(conn.isClosed()).thenReturn(false);
+        DatabaseUtil.closeAllConnections(conn, preparedStatement);
+    }
+
+    @Test
+    public void testClosePreparedStatementHandlesSQLRecoverableException() {
+
+        boolean isThrown = false;
+        try {
+            Mockito.doThrow(new SQLRecoverableException("throw SQLRecoverableException")).when(preparedStatement).close();
+            DatabaseUtil.closeAllConnections(conn, preparedStatement);
+        } catch (SQLRecoverableException e) {
+            // this is expected
+            isThrown = true;
+           Assert.assertTrue(isThrown);

```

Comment:
Similar to the above scenario, this assertion will be always true as the line before is setting the variable isThrown to true.